### PR TITLE
Bump julia to v1.3

### PR DESCRIPTION
**Pull request by siddharthlal25**
This is being done to use the package `ResumableFunctions.jl` for creating generators.

### CODE REVIEW

**Review by mileslucas — APPROVED** [review]

File: src/collection.jl [hunk 1]
```diff
@@ -88,7 +71,7 @@ function fitscollection(basedir::String;
                 hdu isa ImageHDU || continue
                 header_data = read_header(hdu)
                 path = abspath ? Base.abspath(location) : location
-                name = keepext ? filename : first(split(filename, ext))[1:end-1]
+                name = keepext ? filename : first(split(filename, "." * ext))
```
Comment (giordano) [inline src/collection.jl]: Does this work if the file has a funny name like `foo.fits.fits`?

Comment (siddharthlal25) [reply src/collection.jl]: Doing this gives us only `foo`, I hope we don't get something strange like this. :joy:

Comment (giordano) [reply src/collection.jl]: Well, don't expect users to choose sensible names :wink:

Comment (giordano) [reply src/collection.jl]: I didn't try, but doesn't something like
```suggestion
                name = keepext ? filename : join(split(filename, "." * ext)[1:end-1])
```
work?

Comment (siddharthlal25) [reply src/collection.jl:74]: What would you suggest to tackle this case?

Comment (siddharthlal25) [reply src/collection.jl]: Tried something like this:
```
ext = r"fits(\.tar\.gz)?"i
filename = "foo.fits.fits"

join(split(filename, "." * ext)[1:end-1])
```
This gives `foo` only

Edit:
```
split(filename, "." * ext)
```
gives: 
```
Vector{SubString{String}} with 3 elements
"foo"
""
""
```

Comment (mileslucas) [reply src/collection.jl]:
```julia
parse_name(filename, ext::AbstractString, ::Val{false}) = first(rsplit(filename, ext, limit=2))

function parse_name(filename, ext::Regex, ::Val{false})
	idxs = findall(ext, filename)
    return filename[1:first(last(idxs)) - 1]
end

parse_name(filename, ext, ::Val{true}) = filename

# within the parsing function
name = parse_name(filename, "." * ext, Val(keepext))
```

Comment (siddharthlal25) [reply src/collection.jl]: Should we modify the `ext` by adding `$` at the end and then just call `first(split(filename, "." ext))`?

Comment (siddharthlal25) [reply src/collection.jl]: >     return filename[1:first(last(idxs)) - 1]
Shouldn't this be `[1:first(last(idxs)) - 2]`?

Comment (siddharthlal25) [reply src/collection.jl]: No, it should be 1 only.

**Review by giordano — APPROVED** [review]

**Pull request merged**